lib/dry/logic/build/predicate.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Dry
+  module Logic
+    module Build
+      class Predicate < Base
+        def method_missing(method, *args, **kwargs, &block)
+          super unless respond_to_missing?(method)
+          super if Kernel.block_given?
+          super unless kwargs.empty?
+
+          to_predicate(method).curry(*args)
+        end
+
+        def respond_to_missing?(method, *)
+          to_predicate(method)
+        rescue NameError
```

Comment:
I tried replacing this with `respond_to?` and `method_defined?` on both `Predicates` and `Predicates::Methods`. None of them worked. Any ideas on how that could be done?

Reply:
Fixed in 4716391. The only thing working was `Predicates.methods.include?`. `defined_method?` and `respond_to?` failed for dynamically added predicates (using `define_singleton_method` tru `Predicates::Methods.predicate`). `Predicates.methods.include?` was the only way to test for both pre-existing predicates and dynamically added ones.

Reply:
A faster implementation was added in b7d7e88d931850148a9c3bc8d154aa2187de13ae